### PR TITLE
allow relative model to be able to handle time changing antenna patterns

### PR DESCRIPTION
**Pull request by ahnitz**
This updates the relative model to allow it to handle changes in the antenna pattern as a function of time which can greatly affect the PE of very long duration sources, such as those observed by CE and ET. This includes only the polarization factor corrections and not the doppler shift. It is implemented only as a constant correction on the same scale as the frequency binning. Since it is applied in the frequency domain this should not be used with models with higher order modes or wherever there is signfiicant confusion about what time a particular frequency maps to.

### CODE REVIEW

**Comment by ahnitz** [comment]
depends on #3550 and #3549

**Comment by ahnitz** [comment]
This adds an option to the 'relative' model. It is disabled by default (so current behavior). To enable
```
[model]
vary_polarization = 
```

**Comment by ahnitz** [comment]
@dfinstad This also includes a minor fix. You had inadvertently hard coded the waveform to never be used below 10 Hz.

File: pycbc/inference/models/gaussian_noise.py
Comment (cdcapano) [inline pycbc/inference/models/gaussian_noise.py:209]: Thanks for getting rid of the setter, it was needlessly complicated and didn't actually do anything useful. However, please keep `high_frequency_cutoff` as a property that returns `_f_upper` rather than an attribute. I don't want the user thinking they can change the high frequency cutoff by setting that. The reason this isn't settable after the init is the weights would need to be recalculated. By making it an attribute, the user might think they can change it by settting `high_frequency_cutoff`, when it actually doesn't do anything.

I just realized you had also changed `low_frequency_cutoff` from a property to an attribute in an earlier PR. Please switch that back too. So, change `self._f_lower = self.low_frequency_cutoff = low_frequency_cutoff` in the init to `self._f_lower = low_frequency_cutoff`. Then add:
```
@property 
def low_frequency_cutoff(self):
    """The low frequency cutoff of the inner product."""
    return self._f_lower
```
Ditto for `high_frequency_cutoff`.

**Review by cdcapano — CHANGES_REQUESTED** [review]
The projection bit looks fine to me. I have a request for the changes you made to the `BaseGaussianNoise` class though; see below.

File: pycbc/inference/models/relbin.py
Comment (ahnitz) [reply pycbc/inference/models/relbin.py:240]: For technical reasons that doesn't catch everything. Parameters from the configparse in the PE code don't always get transformed to the obvious thing for true/ false statements. i.e. something =, doesn't return true, it gets made a string which won't evaluate quite the way you are thinking.

File: pycbc/inference/models/gaussian_noise.py
Comment (ahnitz) [reply pycbc/inference/models/gaussian_noise.py:209]: @cdcapano Ok, done.

**Comment by ahnitz** [comment]
@cdcapano Please review again

File: pycbc/inference/models/gaussian_noise.py
Comment (cdcapano) [reply pycbc/inference/models/gaussian_noise.py:209]: Thanks!

File: pycbc/inference/models/relbin.py
Comment (cdcapano) [reply pycbc/inference/models/relbin.py:240]: Ok

**Review by cdcapano — APPROVED** [review]
👍

**Comment by cdcapano** [comment]
@ahnitz Code climate is complaining about some whitespace.

**Comment by ahnitz** [comment]
@dfinstad I'll wait on your sign-off as this changes some of your code.

**Review by dfinstad — APPROVED** [review]
The changes look good to me. Thanks for doing this!

**Pull request merged**
